### PR TITLE
Optional questions ep1&2

### LINKED ISSUE #390: Add optional questions to exercises in episode 1 & 2
All exercises in episode 3 & 4 have nice optional questions. This works really well, because the audience is usually quite mixed in terms of their deep learning experience. We should also add them to episode 1 & 2.

### PR DESCRIPTION
**Pull request by svenvanderburg**
Fixes #390

### CODE REVIEW

File: episodes/1-introduction.Rmd [hunk 1]
```diff
@@ -61,15 +61,18 @@ A neural network consists of connected computational units called **neurons**. E
...
 Multiple neurons can be joined together by connecting the output of one to the input of another. These connections are associated with weights that determine the 'strength' of the connection, the weights are adjusted during training. In this way, the combination of neurons and connections describe a computational graph, an example can be seen in the image below. In most neural networks neurons are aggregated into layers. Signals travel from the input layer to the output layer, possibly through one or more intermediate layers called hidden layers.
 The image below shows an example of a neural network with three layers, each circle is a neuron, each line is an edge and the arrows indicate the direction data moves in.
 
 ![](../fig/01_neural_net.png){alt='A diagram of a three layer neural network with an input layer, one hidden layer, and an output layer.'}
 The image above is by Glosser.ca, CC BY-SA 3.0 <https://creativecommons.org/licenses/by-sa/3.0>, via Wikimedia Commons, [original source](https://commons.wikimedia.org/wiki/File:Colored_neural_network.svg)
 
 ::: challenge
-## Calculate the output for one neuron
+## Neural network calculations
+.
```
Comment (svenvanderburg) [inline episodes/1-introduction.Rmd:73]: The . here is necessary to have it actually render the H2.

File: episodes/1-introduction.Rmd [hunk 2]
```diff
@@ -81,16 +84,42 @@ What is the output of the neuron?
...
+| 0  | 1  | ..|
+| 1  | 0  | ..|
+| 1  | 1  | ..|
+
+b. What logical problem does this network solve?
+
+:::: solution
+## Solution
+#### 1: calculate the output for one neuron
+* Weighted sum of input: 0 * (-1) + 0.5 * (-0.5) + 1 * 0.5 = 0.25
```
Comment (psteinb) [inline episodes/1-introduction.Rmd:107]:
```suggestion
* Weighted sum of input: `0 * (-1) + 0.5 * (-0.5) + 1 * 0.5 = 0.25`
```

File: episodes/1-introduction.Rmd [hunk 3]
```diff
@@ -81,16 +84,42 @@ What is the output of the neuron?
...
+* Add the bias: 0.25 + 1 = 1.25
```
Comment (psteinb) [inline episodes/1-introduction.Rmd:108]:
```suggestion
* Add the bias: `0.25 + 1 = 1.25`
```

File: episodes/1-introduction.Rmd [hunk 4]
```diff
@@ -81,16 +84,42 @@ What is the output of the neuron?
...
+* Apply activation function: max(1.25, 0) = 1.25
```
Comment (psteinb) [inline episodes/1-introduction.Rmd:109]:
```suggestion
* Apply activation function: `max(1.25, 0) = 1.25`
```

File: episodes/1-introduction.Rmd [hunk 5]
```diff
@@ -81,16 +84,42 @@ What is the output of the neuron?
...
 
 So, neuron output = 1.25
```
Comment (psteinb) [inline episodes/1-introduction.Rmd:111]:
```suggestion
So, the neuron's output is `1.25`
```

File: episodes/2-keras.Rmd [hunk 6]
```diff
@@ -142,13 +142,27 @@ Take a look at the pairplot we created. Consider the following questions:
...
 
+Answer to optional question:
+
+```python
+sns.pairplot(penguins, hue='sex')
+```
+
+![][sex_pairplot]
+
+You see that for each species the females have smaller bills and flippers, as well as a smaller body mass.
```
Comment (psteinb) [inline episodes/2-keras.Rmd:162]:
```suggestion
You see that for each species females have smaller bills and flippers, as well as a smaller body mass.
```

**Review by psteinb — COMMENTED** [review]
Looks good overall. I mostly have nitpicks regarding formatting. Please let me know what you think.

**Comment by svenvanderburg** [comment]
Yes, good to format the equations like this!

**Pull request merged**